tests/unit/test_metadata.py

```diff
@@ -1,48 +1,109 @@
 """Contains tests to validate metadata.py."""
+from enum import Enum
+
 import pytest
 
 from ni_measurementlink_service._internal.parameter import metadata
-from ni_measurementlink_service.measurement.info import DataType
+from ni_measurementlink_service.measurement.info import DataType, TypeSpecialization
+
+
+class Color(Enum):
```

Comment:
@synesthetize I vaguely remember something about enums having to have a 0 value. Did we have a restriction like this or associated validation?

Reply:
Ah, I overlooked this requirement. I've added a check to validate enums have a zero value and added a test for this. Also fixed up existing tests.